### PR TITLE
Implemented: support to sync the mappings with server, added support to update and delete field mapping, and improved the UI for the functionality(#85zrhcq56)

### LINKED ISSUE #129: Integrate field mappings feature with API
## What is the motivation for adding/enhancing this feature?

Currently, we store field mappings in localstorage, use APIs to store it at backend.

APIs:

createDataManagerMapping
Parameters:
mappingPrefId (optional)
mappingPrefValue
mappingPrefTypeEnumId
fromDate
thruDate

updateDataManagerMapping
Parameters:
mappingPrefId (optional)
mappingPrefValue
mappingPrefTypeEnumId
fromDate
thruDate

deleteDataManagerMapping
Parameters:
mappingPrefId

## What are the acceptance criteria?

- [ ] ...

## Can you complete this feature request by yourself?

- [ ] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by ymaheshwari1**
### Related Issues

Closes #129 

### Screenshots of Visual Changes before/after (If There Are Any)

Saved Mapping:

image

Saved Mapping with configuration component:

image

Purchase Order:

image

Create field mapping model:

image

Mobile View (for configuration component):

image

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/import#contribution-guideline)

### Additional Information:
- For now when deleting a mapping the configuration component does not close, but its values gets cleared.

### CODE REVIEW

**Review by disha1202 — APPROVED** [review]
LGTM 👍🏻

File: src/views/PurchaseOrder.vue [hunk 1]
```diff
@@ -188,32 +165,41 @@ export default defineComponent({
...
+            fieldMapping.value[key] = "";
+          }
+        })
+        this.fieldMapping = fieldMapping.value;
       },
       areAllFieldsSelected() {
         return Object.values(this.fieldMapping).every(field => field !== "");
+      },
+      async addFieldMapping() {
+        const fieldMappingModal = await modalController.create({
```
Comment (adityasharma7) [inline src/views/PurchaseOrder.vue:191]:
```suggestion
        const createMappingModal = await modalController.create({
```

File: src/views/SavedMappings.vue [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      isDesktop: isPlatform('desktop'),
+      isMappingConfigAnimationCompleted: false,
+      currentMapping: {}
+    }
+  },
+  methods: {
+    async viewMappingConfiguration(mapping: any, id: string) {
+      this.currentMapping = {
+        id,
+        ...mapping
```
Comment (adityasharma7) [inline src/views/SavedMappings.vue:83]: I think this might not be needed

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/components/CreateMappingModal.vue [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      }
+      const id = this.generateUniqueMappingPrefId();
+      return this.store.dispatch("user/createFieldMapping", { id, name: this.mappingName, value: this.fieldMapping }).then(() => {
+        this.closeModal()
+      })
+    },
+    areAllFieldsSelected() {
+      return Object.values(this.fieldMapping).every(field => field !== "");
+    },
+    generateUniqueMappingPrefId(): any {
```
[the remark this answers is not in the document]
Comment (ymaheshwari1) [reply src/components/CreateMappingModal.vue:151]: Sir, as the id is required in the api when creating preference, so generating the id here.
Will create an issue to remove this code when the api will be updated.

**Pull request merged**